src/FsAutoComplete/LspServers/AdaptiveServerState.fs

```diff
@@ -49,24 +50,38 @@ type AdaptiveWorkspaceChosen =
   | NotChosen
 
 
+
+[<AutoOpen>]
+module Helpers3 =
+  open FSharp.Compiler.CodeAnalysis.ProjectSnapshot
+
+  type FSharpReferencedProjectSnapshot with
+
+    member x.ProjectFilePath =
+      match x with
+      | FSharpReferencedProjectSnapshot.FSharpReference(snapshot = snapshot) -> snapshot.ProjectFileName |> Some
+      | _ -> None
+
 [<CustomEquality; NoComparison>]
 type LoadedProject =
-  { FSharpProjectOptions: FSharpProjectOptions
+  { ProjectOptions: Types.ProjectOptions
+    FSharpProjectSnapshot: aval<FSharpProjectSnapshot>
```

Comment:
Since snapshots can change over time (See `ProjectWorkspace.fs`), this is an `aval`.